### PR TITLE
[ADD] Main Task creation from activity

### PR DESCRIPTION
**Pull request by eilst**

### CODE REVIEW

File: project_event/models/project_task.py [hunk 1]
```diff
@@ -123,6 +123,18 @@ def create(self, vals):
...
+            vals['parent_id'] = return_create.id
+            vals['message_follower_ids'] = None
+            self.create_main_task(vals)
+        return return_create
+
+    @api.multi
+    def create_main_task(self, vals):
+        vals['activity_task_type'] = 'task'
+        vals['code'] = self.env['ir.sequence'] \
+            .next_by_code('project.task.task')
```
Comment (jananjoy) [inline project_event/models/project_task.py:138]: @eilst : Code  looks redundant. Hope we can still optimize methods
e.g if vals['activity_task_type'] == 'activity':  it simply returns   self.create_main_task(vals) 
 if vals['activity_task_type'] == 'task'::  it simply returns  return_create on execution of def create(self, vals):

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -123,6 +123,18 @@ def create(self, vals):
...
                     .next_by_code('project.task.task')
+        return_create = super(Task, self).create(vals)
+        if vals['activity_task_type'] == 'activity':
+            vals['parent_id'] = return_create.id
+            vals['message_follower_ids'] = None
+            self.create_main_task(vals)
+        return return_create
+
+    @api.multi
+    def create_main_task(self, vals):
```
Comment (jananjoy) [inline project_event/models/project_task.py:135]: @eilst: def create_main_task(self, vals) method funtional logic  is the same  replicate condition of  elif case in the def create(self, vals) methods .

**Review by jananjoy — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py (hunk 2 above)
Comment (eilst) [reply project_event/models/project_task.py:135]: done

File: project_event/models/project_task.py (hunk 1 above)
Comment (eilst) [reply project_event/models/project_task.py:138]: done

File: project_event/models/project_task.py (hunk 2 above)
Comment (rimbendhaou) [reply project_event/models/project_task.py:135]: @eilst the new task is displayed in the activity list of the event

File: project_event/models/project_task.py [hunk 3]
```diff
@@ -120,9 +120,21 @@ def create(self, vals):
...
+                vals['message_follower_ids'] = None
+                self.create_main_task(vals)
             elif vals['activity_task_type'] == 'task':
                 vals['code'] = self.env['ir.sequence'] \
                     .next_by_code('project.task.task')
+                return_create = super(Task, self).create(vals)
+        return return_create
+
+    @api.multi
+    def create_main_task(self, vals):
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:135]: @eilst the new created task is displayed in the activity list of the event

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py (hunk 3 above)
Comment (eilst) [reply project_event/models/project_task.py:135]: done

File: project_event/models/project_task.py (hunk 2 above)
Comment (eilst) [reply project_event/models/project_task.py:135]: done

**Pull request closed without merging**